js/index.js

```diff
@@ -1,10 +1,77 @@
 const isEqual = require("lodash.isequal");
 const expects = require("./expects.js");
 
+function getAllIndexes(arr, val) {
+  var indexes = [], i;
+  for (i = 0; i < arr.length; i++)
+    if (arr[i] === val)
+      indexes.push(i);
+  return indexes;
+}
+
+function checkVertical(puzzle, y, x, word) {
+  for (let i = 0; i < word.length; i++) {
+    if (word[i] !== puzzle[y][x]) {
+      return 0
```

Comment:
I would return a boolean here instead of a number. JavaScript does have many different ways to define truthy values by coercing non-boolean types which can become very confusing https://developer.mozilla.org/en-US/docs/Glossary/Truthy.

I think a boolean makes the most sense